src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> oldTail = tail;
+        Node<T> newNode = new Node<>(oldTail, value, null);
+        tail = newNode;
+        if (oldTail == null) {
+            head = newNode;
+        } else {
+            oldTail.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkElementIndexForAdd(index);
+        Node<T> foundNode = find(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> prev = foundNode.prev;
+            Node<T> newNode = new Node<>(prev, value, foundNode);
+            foundNode.prev = newNode;
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return find(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> findNode = find(index);
+        T oldVal = findNode.item;
+        findNode.item = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> oldNode = find(index);
+        return unlink(oldNode);
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = head; node != null; node = node.next) {
+            if (object == node.item || object != null && node.item.equals(object)) {
```

Comment:
```suggestion
            if (object == node.item || object != null && object.equals(node.item)) {
```
before calling `equals()` on object we should make sure that it is not null